File: microservices/Models/Attendee.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace EventManagementSystem.Models
{
    [Table("Attendees", Schema = "EventManagement")]
    public class Attendee
    {
        [Key]
        public int attendee_id { get; set; }

        public int event_id { get; set; }
        public int user_id { get; set; }

        [StringLength(100)]
        public string? ticket_type { get; set; } = "Standard"; // Changed to string

        public bool check_in_status { get; set; } = false;
    }
}

File: front-end/src/models/attendee.ts
// src/models/attendee.ts

export interface AttendeeResponseDto {
  attendee_id: number;
  event_id: number;
  user_id: number;
  ticket_type: string; // e.g., "Standard", "VIP", "Free"
  check_in_status: boolean;
}

// Frontend interface for displaying attendees
export interface AttendeeForDisplay extends AttendeeResponseDto {
  id: string; // Convert attendee_id to string
  name?: string;              // Fetch from Users API
  email?: string;             // Fetch from Users API
  eventName?: string;         // Fetch from Events API
  registrationDate?: string;
}